File: 백준/Gold/5557. 1학년/1학년.py
N = int(input())
values = list(map(int,input().split()))

dp = [[0]*21 for _ in range(N-1)]
dp[0][values[0]] = 1

#vlues 돌면서 만들 수 있는 거 check
for i in range(1,N-1):
  for j in range(21):
    if dp[i-1][j] > 0:
      if 0<=j+values[i]<21:
        dp[i][j+values[i]]+=dp[i-1][j]
      if 0<=j-values[i]<21:
        dp[i][j-values[i]]+=dp[i-1][j]

print(dp[-1][values[-1]])

File: 백준/Gold/5557. 1학년/1학년.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin','utf8').trim().split('\n')

const N = Number(input.shift())
const values = input[0].trim().split(' ').map(Number)

const dp = [...Array(N-1)].map(()=>Array(21).fill(BigInt(0)))

dp[0][values[0]] = BigInt(1)

for (let [idx,value] of values.entries()){
    if (idx === 0 || idx === N-1) continue
    for (let [n,cnt] of dp[idx-1].entries()){
        if (cnt > 0){
            if (n+value>=0 && n+value <=20) dp[idx][n+value] += BigInt(cnt)
            if (n-value>=0 && n-value <=20) dp[idx][n-value] += BigInt(cnt)
        }
    }
    // console.log(idx,dp[idx])
}
console.log(dp[dp.length-1][values[N-1]].toString())